modules/docs/lib/ExampleCodeBlock.tsx

```diff
@@ -45,6 +53,17 @@ const cardStencil = createStencil({
 
 export const ExampleCodeBlock = ({code}: any) => {
   const [isCodeDisplayed, setCodeDisplayed] = React.useState(false);
+  const textInput = React.useRef(null);
+  const [copied, setCopied] = React.useState(false);
+
+  const onCopy = () => {
+    setCopied(true);
+    // @ts-ignore
+    navigator.clipboard.writeText(textInput.current.textContent);
+    setTimeout(() => {
```

Comment:
we should move this into a useEffect, otherwise we're always creating a new setTimeout and never clearing it, for reference: https://www.freecodecamp.org/news/how-to-use-settimeout-in-react-using-hooks/

```tsx
 React.useEffect(() => {
    if (loadingStatus === 'loading') {
      const mockLoading = setTimeout(() => {
        setLoadingStatus('success');
        setQuote(getQuote());
      }, 3000);

      return () => {
        clearTimeout(mockLoading);
      };
    }
  }, [loadingStatus]);
```

or 
```
const timer = React.useRef<ReturnType<typeof setTimeout>>();
  const [keyTypedSofar, setKeyTypedSofar] = React.useState(keySofar.current);

  const startClearKeysSoFarTimer = () => {
    if (timer.current) {
      clearTimeout(timer.current);
    }
    timer.current = setTimeout(() => {
      keySofar.current = '';
      setKeyTypedSofar(keySofar.current);
    }, 500);
  };
```

Reply:
Updated